tests/concepts/tconcepts_issues.nim

```diff
@@ -114,14 +114,15 @@ block t976:
   # bug #6249
   type
       Obj2 = ref object
+      # Note: since everything is Printable now, this test isn't really that useful anymore.
```

Comment:
Note: Callback types continue not to be printable.

Reply:
I tried to implement that. But I failed at implementing it. If the callback is ``cdecl`` I can cast it to ``pointer`` and print the hex value. But I can't test if the callback actually is a ``cdecl``. For other callbacks I could currently implement an output that signals if they have a value or not (might already be more useful than nothing at all). I don't know if I could use RTTI to print function names. I don't want to rely on it though. After all ``gdb`` is already able to print function names for callbacks. Here that is how it looks in in `gdb`:

```C
#include <stdio.h>

void foo() {
  printf("foo\n");
}

void bar() {
  printf("foo\n");
}

typedef void (*callback)();

void foobar(callback cb) {
  cb();
}

int main() {
  foobar(foo);
  foobar(bar);
}
```

```
(gdb) break foobar
Breakpoint 1 at 0x116b: file scratch.c, line 14.
(gdb) run
Starting program: /tmp/a.out 

Breakpoint 1, foobar (cb=0x555555555139 <foo>) at scratch.c:14
14	  cb();
(gdb) print cb
$1 = (callback) 0x555555555139 <foo>
(gdb) continue
Continuing.

Breakpoint 1, foobar (cb=0x55555555514c <bar>) at scratch.c:14
14	  cb();
(gdb) print cb
$2 = (callback) 0x55555555514c <bar>
(gdb) 
```